### ISSUE TITLE
NotImplementedError because type is Tensor

### ISSUE THREAD

**Issue by jbdel** [locuslab/pytorch_fft#34]
Hello,

I'm using PyTorch 0.4 and I cannot call any of the fft methods, because I always fall into the NotImplementedError case.

Indeed, methods have such codes : 
```
def fft(X_re, X_im): 
    if 'Float' in type(X_re).__name__ :
        f = th_fft.th_Float_fft1
    elif 'Double' in type(X_re).__name__: 
        f = th_fft.th_Double_fft1
    else: 
        raise NotImplementedError
    return _fft(X_re, X_im, f, 1)
```

But the functions args are of type `Tensor` (if i print `type(X_re).__name__) `for example.

I simply tried with the homepage example : 
```
# Example that uses the autograd for 2D fft:
import torch
from torch.autograd import Variable
import pytorch_fft.fft.autograd as fft
import numpy as np

f = fft.Fft2d()
invf= fft.Ifft2d()

fx, fy = (Variable(torch.arange(0,100).view((1,1,10,10)).cuda(), requires_grad=True), 
          Variable(torch.zeros(1, 1, 10, 10).cuda(),requires_grad=True))
k1,k2 = f(fx,fy)
z = k1.sum() + k2.sum()
z.backward()
print(fx.grad, fy.grad)
```
Thanks in advance for your help

**Comment by riceric22** [comment]
Hello! This package hasn't been migrated to PyTorch 0.4.

However, it appears that PyTorch 0.4 has finally added native FFT support, see 

https://pytorch.org/docs/stable/torch.html?highlight=fft#torch.fft

So I'm not sure if there's any value in porting the current package to 0.4, as it seems to only have use it 0.3 and prior versions which don't have native FFT support, unless there's some needed FFT functionality that the PyTorch 0.4 version doesn't have.

**Comment by jbdel** [comment]
Thank you for your response.
I didnt check if fft's where included in 0.4, sorry !

An easy work around for your package is to use tensor.type() (which returns 'torch.FloatTensor') instead of type(tensor).__name__ (which returns only 'Tensor')

Good day,
